app/src/main/java/com/bogdan/codeforceswatcher/fragment/UsersFragment.kt

```diff
@@ -0,0 +1,126 @@
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import com.bogdan.codeforceswatcher.CwApp
+import com.bogdan.codeforceswatcher.R
+import com.bogdan.codeforceswatcher.activity.AddUserActivity
+import com.bogdan.codeforceswatcher.adapter.UserAdapter
+import com.bogdan.codeforceswatcher.model.User
+import com.bogdan.codeforceswatcher.util.Prefs
+import com.bogdan.codeforceswatcher.util.UserLoader
+import kotlinx.android.synthetic.main.fragment_users.*
+
+class UsersFragment : android.support.v4.app.Fragment(), SwipeRefreshLayout.OnRefreshListener, View.OnClickListener {
+
+    private lateinit var userAdapter: UserAdapter
+    private var counterIcon: Int = 0
+
+    override fun onRefresh() {
+        UserLoader.loadUsers(CwApp.app.userDao.getAll()) { activity?.runOnUiThread { swiperefresh.isRefreshing = false } }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        activity?.let {
+            val prefs = Prefs(it)
+            counterIcon = if (prefs.readCounter().isEmpty()) 0 else prefs.readCounter().toInt()
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_users, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initViews()
+    }
+
+    private fun initViews() {
+        val fab = activity?.findViewById<FloatingActionButton>(R.id.fab)
+        val spSort = activity?.findViewById<AppCompatSpinner>(R.id.spSort)
+        fab?.visibility = View.VISIBLE
+        spSort?.visibility = View.VISIBLE
+        fab?.setOnClickListener(this)
+        swiperefresh.setOnRefreshListener(this)
+
+        val spinnerAdapter = ArrayAdapter(activity, android.R.layout.simple_list_item_1,
+                resources.getStringArray(R.array.array_sort))
+        spinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+
+        spSort?.adapter = spinnerAdapter
+
+        spSort?.setSelection(counterIcon)
+
+        spSort?.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
+                counterIcon = position
+                updateList(CwApp.app.userDao.getAll())
+                activity?.let {
+                    val prefs = Prefs(it)
+                    prefs.writeCounter(counterIcon)
+                }
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>) {
+            }
+        }
+        activity?.let {
```

Comment:
You can pass non optional activity into `initViews` method as a param to get rid of all that `?`.